Title: change Makefile to comply with Debian
Author: Tong Sun <suntong001@users.sourceforge.net>
Last-Update: 2014-08-10
DEP: 3
Forwarded: no
Update: modify to match new file name: makefile
Author: Javier Fafian Alvarez <fafian@gmail.com>
Last-Update: 2015-09-04

Index: shc-3.8.9/makefile
===================================================================
--- shc-3.8.9.orig/makefile
+++ shc-3.8.9/makefile
@@ -1,85 +1,52 @@
-# Makefile
-#
+INSTALL = /usr/bin/install -c
 
-INSTALL_PATH = /usr/local
+# == Installation directories
+prefix = ${DESTDIR}/usr
+mandir = ${prefix}/share/man/man1
+exec_prefix = ${prefix}
+bindir = ${exec_prefix}/bin
+docdir = ${prefix}/share/doc/shc
+
+# == Build directives
+## For SCO
+#CFLAGS = -b elf -O -D_SVID
+## For IRIX
+#CFLAGS = -xansi -fullwarn -O3 -g0
+## For Solaris
+#CFLAGS = -fast -xO4 -s -v -Xa
+## For HPUX
+#CFLAGS = -Wall -O -Ae
+## For OSF1
+#CFLAGS = -w -verbose -fast -std1 -g0
+## For GNU C compiler
+#CFLAGS = -Wall # -O6 -pedantic
 
-# For SCO
-CFLAGS = -b elf -O -D_SVID
-
-# For IRIX
-CFLAGS = -xansi -fullwarn -O3 -g0
-
-# For Solaris
-CFLAGS = -fast -xO4 -s -v -Xa
-
-# For HPUX
-CFLAGS = -Wall -O -Ae
-
-# For OSF1
-CFLAGS = -w -verbose -fast -std1 -g0
-
-# For GNU C compiler
-CFLAGS = -Wall # -O6 -pedantic
-
-#SHELL = /bin/sh
-
-all: shc ask_for_test
+all: shc
 
 shc: shc.c
-	$(CC) $(CFLAGS) $@.c -o $@
-
-ask_for_test:
-	@echo '***	¿Do you want to probe shc with a test script?'
-	@echo '***	Please try...	make test'
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $@.c -o $@
 
-test: make_the_test ask_for_strings
+test: make_the_test
 
 make_the_test: match.x
 	@echo '***	Running a compiled test script!'
 	@echo '***	It must show files with substring "sh" in your PATH...'
-	./match.x sh
+	./match.x sh || true
 
 match.x: shc match
 	@echo '***	Compiling script "match"'
-	CFLAGS="$(CFLAGS)" ./shc -v -f match
-
-ask_for_strings:
-	@echo '***	¿Do you want to see strings in the generated binary?'
-	@echo '***	Please try...	make strings'
-
-strings: make_the_strings ask_for_expiration
-
-make_the_strings: match.x
-	@echo '***	Running: "strings -n 5 'match.x'"'
-	@echo '***	It must show no sensible information...'
-	strings -n 5 match.x
-
-ask_for_expiration:
-	@echo '***	¿Do you want to probe expiration date?'
-	@echo '***	Please try...	make expiration'
-
-expiration: til_yesterday ask_for_install
-
-til_yesterday: shc match
-	@echo '***	Compiling "match" to expired date'
-	CFLAGS="$(CFLAGS)" ./shc -vvv -e `date "+%d/%m/%Y"` -f match
-	@echo '***	Running a compiled test script!'
-	@echo '***	It must fail showing "./match.x: has expired!"'
-	./match.x
-
-ask_for_install:
-	@echo '***	¿Do you want to install shc?'
-	@echo '***	Please try...	make install'
-
-install: shc
-	@echo '***	Installing shc and shc.1 on '$(INSTALL_PATH)
-	@echo -n '***	¿Do you want to continue? '; read ANS; case "$$ANS" in y|Y|yes|Yes|YES) ;; *) exit 1;; esac;
-	install -c -s shc $(INSTALL_PATH)/bin/
-	install -c -m 644 shc.1 $(INSTALL_PATH)/man/man1/
 
 clean:
-	rm -f *.o *~ *.x.c
+	rm -f *.o *~ *.x.c shc *.x
 
 cleanall: clean
 	rm -f shc *.x
 
+install:
+	$(INSTALL) -m 755 -d $(bindir)
+	$(INSTALL) -m 755 -d $(mandir)
+	$(INSTALL) -m 755 -d $(docdir)
+
+	$(INSTALL) -m 755 -s shc $(bindir)
+	$(INSTALL) -m 644 shc.1 $(mandir)
+	$(INSTALL) -m 644 match* shc.html shc.README test.* CHANGES pru.sh $(docdir)
